tests/Doctrine/Tests/ORM/Hydration/ScalarColumnHydratorTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Hydration;
+
+use Doctrine\Tests\Models\CMS\CmsUser;
+use Doctrine\ORM\Query\ResultSetMapping;
+use Doctrine\Tests\Mocks\HydratorMockResult;
+use Doctrine\ORM\Internal\Hydration\ScalarColumnHydrator;
+
+use function count;
+use function is_array;
+
+class ScalarColumnHydratorTest extends HydrationTestCase
+{
+    /**
+     * Select u.id from CmsUser u
+     */
+    public function testEmptyResultTest(): void
+    {
+        $rsm = new ResultSetMapping();
+
+        $rsm->addEntityResult(CmsUser::class, 'u');
+        $rsm->addFieldResult('u', 'u__id', 'id');
+
+        $stmt = new HydratorMockResult($emptyResultSet = []);
+        $hydrator = new ScalarColumnHydrator($this->entityManager);
+        $result = $hydrator->hydrateAll($stmt, $rsm);
+
+        $this->assertTrue(is_array($result));
+        $this->assertEquals(0, count($result));
+    }
+
+    /**
+     * Select u.id from CmsUser u
+     */
+    public function testSingleColumnEntityQueryWithoutScalarMap(): void
+    {
+        $rsm = new ResultSetMapping();
+
+        $rsm->addEntityResult(CmsUser::class, 'u');
+        $rsm->addFieldResult('u', 'u__id', 'id');
+
+        // Faked result set
+        $resultSet = [
+            0 => '1',
+            1 => '2',
+        ];
+
+        $stmt = new HydratorMockResult($resultSet);
+        $hydrator = new ScalarColumnHydrator($this->entityManager);
+        $result = $hydrator->hydrateAll($stmt, $rsm);
+
+        $this->assertTrue(is_array($result));
+        $this->assertEquals(2, count($result));
+
+        $this->assertEquals(1, $result[0]);
+        $this->assertEquals(2, $result[1]);
+    }
+
+    /**
+     * Select u.id from CmsUser u
+     */
+    public function testSingleColumnEntityQueryWithScalarMap(): void
+    {
+        $rsm = new ResultSetMapping();
+
+        $rsm->addEntityResult(CmsUser::class, 'u');
+        $rsm->addFieldResult('u', 'u__id', 'id');
+        $rsm->addScalarResult('sclr0', 'id');
+        $rsm->addIndexByScalar('sclr0');
+
+        // Faked result set
+        $resultSet = [
+            [
+                'u__id' => '1',
+            ],
+            [
+                'u__id' => '2',
+            ],
+        ];
+
+        $stmt = new HydratorMockResult($resultSet);
+        $hydrator = new ScalarColumnHydrator($this->entityManager);
+        $result = $hydrator->hydrateAll($stmt, $rsm);
+
+        $this->assertEquals(2, count($result));
+        $this->assertTrue(is_array($result));
+
+        $this->assertEquals(1, $result[0]);
+        $this->assertEquals(2, $result[1]);
+    }
+
+    /**
+     * SELECT PARTIAL u.{id, name}
+     *   FROM Doctrine\Tests\Models\CMS\CmsUser u
+     */
+    public function testMultipleColumnEntityQueryWithoutScalarMap(): void
```

Comment:
Maybe we should throw an exception when the user selects more than one field instead? What is the point of selecting more than one field, when in the end you only get one?

Reply:
Updated